Support .NET 8 only analyzers as references

DacFX 170 does no longer support netstandard2.0 - wonder if we can make this work with new analyzer packages during build?

I get this error, if I target net8.0:

1>C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets(4905,5): error MSB4044: The "CopyRefAssembly" task was not given a value for the required parameter "SourcePath".

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -186,7 +186,7 @@ jobs:
       uses: actions/upload-artifact@v4
       with:
         name: dacpac-package
-        path: ./test/TestProjectWithSDKRef/bin/Debug/netstandard2.0/
+        path: ./test/TestProjectWithSDKRef/bin/Debug/net8.0/
 
     # Upload binary log
     - name: upload

--- a/src/MSBuild.Sdk.SqlProj.Templates/templates/sqlproj/sqlproj.csproj
+++ b/src/MSBuild.Sdk.SqlProj.Templates/templates/sqlproj/sqlproj.csproj
@@ -1,6 +1,6 @@
 <Project Sdk="MSBuild.Sdk.SqlProj/#{NBGV_NuGetPackageVersion}#">
     <PropertyGroup>
-        <TargetFramework>netstandard2.1</TargetFramework>
+        <TargetFramework>net8.0</TargetFramework>
         <SqlServerVersion>#{SqlServerVersion}</SqlServerVersion>
         <RunSqlCodeAnalysis>#{CodeAnalysis}</RunSqlCodeAnalysis>
         <!-- For additional properties that can be set here, please refer to https://github.com/rr-wfm/MSBuild.Sdk.SqlProj#model-properties -->

--- a/src/MSBuild.Sdk.SqlProj/Sdk/Sdk.props
+++ b/src/MSBuild.Sdk.SqlProj/Sdk/Sdk.props
@@ -26,6 +26,7 @@
     <EnableDefaultCompileItems>false</EnableDefaultCompileItems>
     <EnableDefaultEmbeddedResourceItems>false</EnableDefaultEmbeddedResourceItems>
     <NoPackageAnalysis>True</NoPackageAnalysis>
+    <ProduceReferenceAssembly>false</ProduceReferenceAssembly>
     <NuspecFile Condition="Exists('$(MSBuildProjectName).nuspec')">$(MSBuildProjectName).nuspec</NuspecFile>
   </PropertyGroup>
 

--- a/test/TestProject/TestProject.csproj
+++ b/test/TestProject/TestProject.csproj
@@ -2,7 +2,7 @@
   <Import Project="$(MSBuildThisFileDirectory)..\..\src\MSBuild.Sdk.SqlProj\Sdk\Sdk.props" />
 
   <PropertyGroup>
-    <TargetFramework>netstandard2.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
     <SqlServerVersion>Sql150</SqlServerVersion>
     <RecoveryMode>Simple</RecoveryMode>
     <AllowSnapshotIsolation>True</AllowSnapshotIsolation>

--- a/test/TestProjectWithAnalyzers/TestProjectWithAnalyzers.csproj
+++ b/test/TestProjectWithAnalyzers/TestProjectWithAnalyzers.csproj
@@ -2,7 +2,7 @@
   <Import Project="$(MSBuildThisFileDirectory)..\..\src\MSBuild.Sdk.SqlProj\Sdk\Sdk.props" />
 
   <PropertyGroup>
-    <TargetFramework>netstandard2.1</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
     <SqlServerVersion>Sql150</SqlServerVersion>
     <CodeAnalysisRules>-SqlServer.Rules.SRD0006;-Smells.*;+!SqlServer.Rules.SRN0002</CodeAnalysisRules>
     <RunSqlCodeAnalysis>true</RunSqlCodeAnalysis>
@@ -13,7 +13,7 @@
   </ItemGroup>
 
     <ItemGroup>
-        <PackageReference Include="ErikEJ.DacFX.SqlServer.Rules" Version="1.1.2">
+        <PackageReference Include="ErikEJ.DacFX.SqlServer.Rules" Version="3.0.0-preview.2">
           <PrivateAssets>all</PrivateAssets>
           <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
         </PackageReference>

--- a/test/TestProjectWithSDKRef/TestProjectWithSDKRef.csproj
+++ b/test/TestProjectWithSDKRef/TestProjectWithSDKRef.csproj
@@ -1,6 +1,6 @@
 <Project Sdk="MSBuild.Sdk.SqlProj/#{NBGV_NuGetPackageVersion}#">
     <PropertyGroup>
-        <TargetFramework>netstandard2.0</TargetFramework>
+        <TargetFramework>net8.0</TargetFramework>
     </PropertyGroup>
 
     <ItemGroup>

